# Level 10

[Question](http://www.pythonchallenge.com/pc/return/bull.html)

There are two bulls on the question page and a sentence: **len(a[30])?**. 
Open the page source and there is a link for [sequence.txt](http://www.pythonchallenge.com/pc/return/sequence.txt). You need to input user name and password if you open the file for the first time. Actually the user name and password is the same as in Question8. 

Username: huge
Password: file

And I guess if a question needs a file to give us a hint, the password will always be it. 

Sequence.txt shows a unfinished list as below
**[1, 11, 21, 1211, 111221, ...**
This is only the first five object of the list and we need to find the rule. If you are fimiliar with math you will find out that this sequence is called [look and say sequence](https://en.wikipedia.org/wiki/Look-and-say_sequence). 

> - 1 is read off as "one 1" or 11.
> - 11 is read off as "two 1s" or 21.
> - 21 is read off as "one 2, then one 1" or 1211.
> - 1211 is read off as "one 1, one 2, then two 1s" or 111221.
> - 111221 is read off as "three 1s, two 2s, then one 1" or 312211.

So we need to write a method to generate the number in this sequence. It is pretty easy.

## Solution

In [1]:
# a = [1, 11, 21, 1211, 111221,


# this is a look-and-say sequence

import re

def describe(s):
    sequence = re.findall('1+|2+|3+', s)
    return ''.join(str(len(x)) + x[0] for x in sequence)

s = '1'
for i in range(30):
    s = describe(s)

print len(s)


5808


So the anwser is **5808**. Here is the [link]() to next level. 